pkg/networkpolicy/networkpolicy.go

```diff
@@ -0,0 +1,311 @@
+	}
+
+	return builder
+}
+
+// WithNamespaceIngressRule ingress rule for the networkPolicy.
+func (builder *NetworkPolicyBuilder) WithNamespaceIngressRule(namespaceIngressMatchLabels map[string]string,
+	podIngressMatchLabels map[string]string) *NetworkPolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Applying Ingress rule to networkPolicy %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if namespaceIngressMatchLabels == nil && podIngressMatchLabels == nil {
+		glog.V(100).Infof("At least one type of the selector for NetworkPolicy ingress rule should be defined")
+
+		builder.errorMsg = "Both namespaceIngressMatchLabels and podIngressMatchLabels parameters are empty maps"
+
+		return builder
+	}
+
+	var peerRule netv1.NetworkPolicyPeer
+
+	if namespaceIngressMatchLabels != nil {
+		glog.V(100).Infof(
+			"Applying Ingress rule with namespaceIngressMatchLabels %v parameter to networkPolicy %s in namespace %s",
+			namespaceIngressMatchLabels, builder.Definition.Name, builder.Definition.Namespace)
+
+		peerRule.NamespaceSelector = &metav1.LabelSelector{
+			MatchLabels: namespaceIngressMatchLabels,
+		}
+	}
+
+	if podIngressMatchLabels != nil {
+		glog.V(100).Infof(
+			"Applying Ingress rule with podIngressMatchLabels %v parameter to networkPolicy %s in namespace %s",
+			podIngressMatchLabels, builder.Definition.Name, builder.Definition.Namespace)
+
+		peerRule.PodSelector = &metav1.LabelSelector{
+			MatchLabels: podIngressMatchLabels,
+		}
+	}
+
+	if builder.Definition.Spec.Ingress == nil {
+		builder.Definition.Spec.Ingress = make([]netv1.NetworkPolicyIngressRule, 1)
+	}
+
+	ingressRule := netv1.NetworkPolicyIngressRule{}
+	ingressRule.From = append(ingressRule.From, peerRule)
+
+	builder.Definition.Spec.Ingress = append(builder.Definition.Spec.Ingress, ingressRule)
+
+	return builder
+}
+
+// WithPolicyType add policyType to the networkPolicy.
+func (builder *NetworkPolicyBuilder) WithPolicyType(policyType netv1.PolicyType) *NetworkPolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Creating networkPolicy %s with the policyType defined: %d", builder.Definition.Name)
+
+	if policyType == "" {
+		glog.V(100).Infof("The policyType value have to be provided")
+
+		builder.errorMsg = "The policyType is empty sting"
+
+		return builder
+	}
+
+	builder.Definition.Spec.PolicyTypes = append(builder.Definition.Spec.PolicyTypes, policyType)
+
+	return builder
+}
+
+// WithPodSelector add podSelector to the networkPolicy.
+func (builder *NetworkPolicyBuilder) WithPodSelector(podSelectorMatchLabels map[string]string) *NetworkPolicyBuilder {
+	if valid, err := builder.validate(); !valid {
+		builder.errorMsg = err.Error()
+
+		return builder
+	}
+
+	glog.V(100).Infof(
```

Comment:
Please add podSelectorMatchLabels parameter to the log message using %v formatter